lms/djangoapps/instructor/tests/test_api.py

```diff
@@ -136,6 +139,405 @@ def test_alreadyrunningerror_ajax(self):
+        saved_queries = self._get_saved_queries()
+        self.assertEquals(len(saved_queries), 2)
+        query_group = saved_queries[0]['group']
+        self._delete_saved_query(query_group)
+        post_delete_queries = self._get_saved_queries()
+        self.assertEquals(len(post_delete_queries), 0)
+
+    def test_get_group_query(self):
+        """
+        Ensures the logic for getting students associated with a group query
+        """
+
+        self._make_query(self.OPEN)
+        self._make_query(self.NOT_COMPLETED)
+        temp_ids = self._get_temp_query_ids()
+        self._save_query(temp_ids)
+        saved_queries = self._get_saved_queries()
+        self.assertEquals(len(saved_queries), 2)
+        query_group = saved_queries[0]['group']
+        students = get_group_query_students(self.course_key, query_group)
+        correct_students = set()
+        for student in [self.open_user, self.completed_user, self.not_opened_user]:
+            correct_students.add(student.profile.name)
+        check_students = set()
+        for student in students:
+            check_students.add(student['profileName'])
+        self.assertSetEqual(correct_students, check_students)
+
+    def test_purge(self):
+        """
+        Ensures the deletion of old temporary queries
+        """
+
+        self._make_query(self.OPEN)
+        self._make_query(self.COMPLETED)
+        query_obj = TemporaryQuery.objects.all()
+        self.assertNotEqual(len(query_obj), 0)
+        minutes60ago = datetime.datetime.now() - datetime.timedelta(minutes=TEMPORARY_QUERY_LIFETIME * 2)
+        minutes15ago = datetime.datetime.now() - datetime.timedelta(minutes=TEMPORARY_QUERY_LIFETIME / 2)
+        query_obj[0].created = minutes60ago
+        query_obj[1].created = minutes15ago
+        query_obj[0].save()
+        query_obj[1].save()
+        purge_temporary_queries()
+        query_obj = TemporaryQuery.objects.all()
+        self.assertEquals(len(query_obj), 1)
+
+
+@override_settings(MODULESTORE=TEST_DATA_MIXED_MODULESTORE)
+class TestCourseTreeLookup(ModuleStoreTestCase, LoginEnrollmentTestCase):
+    """
+    Ensure appropriate access and format of course information
+    """
+
+    def setUp(self):
+        self.course = CourseFactory.create()
+        course_id = 'edX/emailToy/2015_Winter'
+        self.course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+        self.instructor = InstructorFactory(course_key=self.course_key)
+        CourseEnrollment.enroll(self.instructor, self.course_key)
+        self.client.login(username=self.instructor.username, password='test')
+
+    def test_list_course_tree(self):
+        """
+        Ensures a user can extract the course tree
+        """
+
+        url = reverse('list_course_tree', kwargs={'course_id': self.course_key})
+        stuff = self.client.get(url)
+        response = json.loads(stuff.content)['data']
+        block_ids = set([u'8731f6563d4c4753932d6ad1a22d855a',
```

Comment:
Define the `set` directly, instead of converting from an array:

```python
block_ids = {
    u'8731f6563d4c4753932d6ad1a22d855a',
    u'8dbd29923f894b60b690e38d50807cde',
}
```